docs/xenit-style-guide/golang.md

```diff
@@ -385,3 +334,83 @@ func ginlogrWithSpan(logger logr.Logger) gin.HandlerFunc {
     }
 }
 ```
+
+## HTTP
+
+The Go standard library includes a http package that works well for simple applications, but requires a lot of custom code when building larger projects. While it is fine to just use the standard library for simple applications it is preferable to switch to [Gin](https://github.com/gin-gonic/gin) as project feature requirements develop. Gin provides extra functionality and extensions to simplify things like parsing parameters in the URL path and endpoint authorization.
+
+The HTTP server should be started and stopped in accordance to the Startup and Shutdown documentation. The HTTP server should be gracefully stopped before the program exits to make sure that all in flight requests are processed.
```

Comment:
Link to documentation

Reply:
Will fix in the future as the docs does not yet exist.